graphql/database/widgets/widgets.js

```diff
@@ -0,0 +1,265 @@
+import BaseModel from '../base/model';
+import database from '../database';
+import tablesNames from '../tables';
+
+import Async from 'asyncawait/async';
+import Await from 'asyncawait/await';
+
+import { logger } from '../../utils/dev-tools';
+
+/*
+ * Represents a Base Widget. 
+ * @extends BaseModel
+ */
+class Widget extends BaseModel {
+
+  /**
+   * Creates a Widget instance.
+   * @param {string} id - The instance id in the database.
+   */
+  constructor(id) {
+    super(id);
+
+    this.name = '';
```

Comment:
The `name` field on the widgets seems redundant when we have the `type` and we'll localize the name anyway. Remove?